BraintreeCore/src/main/java/com/braintreepayments/api/core/GetReturnLinkTypeUseCase.kt

```diff
@@ -0,0 +1,34 @@
+package com.braintreepayments.api.core
+
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.annotation.RestrictTo
+
+/**
+ * Use case that returns a return link type that will be used for navigating from App Switch or browser back into the
+ * merchant app.
+ *
+ * If a user unchecks the "Open supported links" checkbox in the Android OS settings for the merchant's app. If this
+ * setting is unchecked, this use case will return [ReturnLinkTypeResult.DEEP_LINK], otherwise
+ * [ReturnLinkTypeResult.APP_LINK] will be returned.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class GetReturnLinkTypeUseCase(private val merchantRepository: MerchantRepository) {
+
+    enum class ReturnLinkTypeResult {
+        APP_LINK, DEEP_LINK
+    }
```

Comment:
Since this enum is exactly the same as the existing `LinkType` enum, would it make sense to just return a `LinkType` from from use case?

Reply:
I think there's some benefit in keeping separation between the analytics specific `LinkType` and the result type of this use case. For example the `GetReturnLinkUseCase` relies on the `GetReturnLinkTypeUseCase` but does not need the analytic specific `LinkType.stringValue` String.

I'm also open to collapsing them down to a single enum, but the above was my thought process in keeping them separate.